frontend/src/features/properties/map/research/add/SelectedPropertyRow.tsx

```diff
@@ -21,7 +21,28 @@ export const SelectedPropertyRow: React.FunctionComponent<ISelectedPropertyRowPr
   index,
   property,
 }) => {
-  const propertyIdentifier = getPropertyIdentifier(property);
+  const propertyName = getPropertyName(property);
+  let propertyIdentifier = '';
+  switch (propertyName.label) {
+    case NameSourceType.PID:
```

Comment:
you can collapse all of the cases to reduce code duplication.

Reply:
sure can do